### PR TITLE
Add rbush v4.x typings

### PR DESCRIPTION
**Pull request by mfedderly**
Follow up to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70242 which works for 3.x but not 4.x with the update from the underlying library to target modules now.

After a quick correction to the underlying library's package.json, the typings worked out to be pretty straightforward.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/rbush/index.d.mts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eac82cb97034dda633582dbf46a9a73305965474
</p>

File: types/polygon-lookup/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     ],
     "dependencies": {
         "@types/geojson": "*",
-        "@types/rbush": "*"
+        "@types/rbush": "^3"
```
Comment (mfedderly) [inline types/polygon-lookup/package.json:10]: The underlying library actually uses 2.x but they've been operating against 3.x for some time now, so I'm a little hesitant to change it. We could bump this down to 2.x and revert this change from the most recent 3.x if we wanted:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70242/files#diff-a07aa4d367a37d9ab6a3042a6c8b663606c397c07601e490a570511f92b61f09R2

File: types/rbush/index.d.mts [hunk 2]
```diff
@@ -0,0 +1,184 @@
+export interface BBox {
```
Comment (mfedderly) [inline types/rbush/index.d.mts:1]: This entire file is very similar but slightly different to index.d.ts, is there an easy way to deduplicate it?

File: types/which-polygon/package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     ],
     "dependencies": {
         "@types/geojson": "*",
-        "@types/rbush": "*"
+        "@types/rbush": "^3"
```
Comment (mfedderly) [inline types/which-polygon/package.json:10]: Same remark as polygon-lookup, the underlying library is actually using rbush 2.x.

**Review by mfedderly — COMMENTED** [review]
First time making a new major version in DT, would appreciate a close look on that change specifically.

File: types/rbush/index.d.mts (hunk 2 above)
Comment (jakebailey) [reply types/rbush/index.d.mts:1]: Yes, you can actually just do `export * from "./index.js` or vice versa.

**Comment by jakebailey** [comment]
I don't think the types here are correct; here is the package.json for `rbuch`:

```
  "type": "module",
  "exports": "./index.js",
  "main": "rbush.js",
  "module": "index.js",
  "browser": "rbush.min.js",
  "jsdelivr": "rbush.min.js",
  "unpkg": "rbush.min.js",
```

This implies that this package needs to be setting `"type": "module"` as well, `index.d.ts` should be ESM, etc

**Comment by mfedderly** [comment]
I attempted to make this `"type": "module"` for a while, but I was unable to get attw to pass. Despite that package declaring itself as a module, the entry for `main` is a commonjs file `rbush.js` with the equivalent of `module.exports = RBush` in it. I may just be not understanding how to change the node10 resolution though.

**Comment by jakebailey** [comment]
if rbush has written CJS code inside of a `js` file when `package.json` contains `"type": "module"`, then that code will actually crash at runtime and it's probably not worth trying to make work.

https://publint.dev/rbush@4.0.0 shows this error.

**Comment by mfedderly** [comment]
I reached out to hopefully correct the underlying package and then I'll return to this PR. Thanks for the help!

**Comment by jakebailey** [comment]
I posted on https://github.com/mourner/rbush/issues/139 but I'm pretty sure 4.0.1 is still broken.

File: types/polygon-lookup/package.json (hunk 1 above)
Comment (andrewbranch) [reply types/polygon-lookup/package.json:10]: Yeah, this range really should match the range in the underlying library. I do think it's probably worth fixing it up now, and dealing with any fallout in a more accurate way if necessary, but these are not super widely depended upon packages, so I feel like it will be fine.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Comment by mfedderly** [comment]
I tried to convert `polygon-lookup` and `which-polygon` to correctly use rbush@^2 but the types were incompatible enough that it seems not worth doing, and especially not worth doing in the rbush@^4 PR. I suspect that the rbush@2 types are actually just wrong in a way that would require (breaking) changes for the dependent packages to even compile.

**Comment by jakebailey** [comment]
That's fair. Maybe write `<=4` for these, then? Rather than pinning on to v3?

**Comment by jakebailey** [comment]
Eh, it's fine, I guess.

**Review by jakebailey — APPROVED** [review]

**Comment by mfedderly** [comment]
Ready to merge

@jakebailey thanks for all the attention here!
